ingest-app/src/cmr/ingest/validation/generic_document_validation.clj

```diff
@@ -0,0 +1,166 @@
+    (mdb/find-concepts context {:latest true} concept-type)
+    (catch Exception e
+      (error (str "Error fetching concepts for generic validations: " (.getMessage e)))
+      []))) 
+
+(defn- validate-uniqueness
+  "Validates that the combination of field values is unique in the collection.
+   Returns a sequence of error messages if validation fails, empty sequence otherwise."
+  [context concept fields]
+  (let [concept-type (:concept-type concept)]
+    (if-let [existing-concepts (fetch-existing-concepts context concept-type)]
+      (let [;; Helper function to extract field values from a concept
+            get-field-values (fn [c fs]
+                               (let [metadata (json/parse-string (:metadata c) true)]
+                                 (mapv #(let [field-path (rest (string/split % #"\."))
+                                              keyword-path (mapv keyword field-path)]
+                                          (get-in metadata keyword-path))
+                                       fs)))
+
+            ;; Extract values for the specified fields from the current concept
+            field-values (get-field-values concept fields)
+
+            ;; Check if any other document in the collection has the same combination of values
+            duplicate-concepts (filter (fn [existing-concept]
+                                         (and (not= (:native-id existing-concept) (:native-id concept))
+                                              (not= (:deleted existing-concept) true)
+                                              (= (set (get-field-values existing-concept fields))
+                                                 (set field-values))))
+                                       existing-concepts)]
+        (if (seq duplicate-concepts)
+          (let [duplicate-concept-ids (map :concept-id duplicate-concepts)
+                field-names (mapv #(string/join "." (rest (string/split % #"\."))) fields)
+                display-values (mapv str field-values)]
+            (info "Duplicate concept IDs found: " duplicate-concept-ids)
+            [(format "Values %s for fields %s must be unique for concept type %s. Duplicate concept IDs: %s"
+                     (string/join ", " display-values)
+                     (string/join ", " field-names)
+                     (str concept-type)
+                     (string/join ", " duplicate-concept-ids))])
+          []))
+      [])))
+
+(defn- validate-by-type
+  "Validates fields based on validation type. Any new validation added to the config.json
+   should be added here and implemented in the corresponding function. 
+   validation-value is here for possible future implementations but is not needed in the 
+   uniqueness validation.
+   Returns a sequence of error messages if validation fails, empty sequence otherwise."
+  [context concept validation-type fields validation-value]
+  (case validation-type
+    "unique" (validate-uniqueness context concept fields)
```

Comment:
consider a case insensitive compare. I don't trust JSON editors to do the right thing.

Reply:
Correct.